backend/src/modules/tournament-tree/core/domain/TournamentTree.ts

```diff
@@ -74,22 +87,26 @@ export class TournamentTree {
     return this.tournamentTreeArray.find((match) => match.matchNumberInSequence === matchNumberInSequence)?.fightingTeamsGroupId.raw;
   }
 
-  public saveMatchResult(matchId: string, winnerTeamId: string): void {
-    const match = this.tournamentTreeArray.find((match) => matchId == match.fightingTeamsGroupId.raw);
-    const nextMatchIdForWinner = match?.nextMatchId?.raw;
+  public finishMatch(finishedMatchId: string, winnerTeamId: string): void {
```

Comment:
![image](https://user-images.githubusercontent.com/18637116/111982628-793dec80-8b09-11eb-8f4e-eb37dcdb2d6f.png)

Ja to pociągnę już tutaj.
Chodzi o to, że jesteśmy w domain. Czyli nasze TournamentTree powinno nie mieć pojęcia, że jest coś takiego jak matchId, który nie jest matchId - bo dla niego to albo matchId albo nie. 
Dlatego do obiektu domenowego - nie przekazujemy stringa, który może być czykolwiek, chociażby "shakalaka". Tylko już tutaj powinny wejśc obiekty typu MatchId i TournamentTeamId, co teraz jest robione później.